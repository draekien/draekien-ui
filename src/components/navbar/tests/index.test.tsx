/** @jsxImportSource theme-ui */

import { fireEvent, render } from '@testing-library/react';
import Navbar from '..';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

describe('Navbar component', () => {
  test('should match snaptshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar backgroundColor="primary" />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('should render mobile menu when viewport width is less than breakpoint', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );

    act(() => {
      global.innerWidth = 990;
      global.dispatchEvent(new Event('resize'));
    });

    expect(container.textContent).toContain('menu');
  });

  test('should open menu when menu is pressed', () => {
    const fn = jest.fn();

    const { container } = render(
      <BrowserRouter>
        <Navbar onClick={fn} />
      </BrowserRouter>
    );

    fireEvent(
      container.querySelector('button') as Element,
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );

    expect(fn).toHaveBeenCalledTimes(1);
  });
});
