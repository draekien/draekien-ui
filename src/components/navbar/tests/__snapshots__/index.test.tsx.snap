// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match snaptshot 1`] = `
<div>
  <div>
    <nav
      class="css-8lvt6e-Navbar"
    >
      <div
        class="css-1sdsjsc-Navbar"
      >
        <a
          href="/"
        />
        <div
          class="css-1vlz94w"
        />
      </div>
    </nav>
  </div>
</div>
`;
